actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -173,6 +173,34 @@ def _filtered_referrer # :nodoc:
       #   link_to "External link", "http://www.rubyonrails.org/", target: "_blank", rel: "nofollow"
       #   # => <a href="http://www.rubyonrails.org/" target="_blank" rel="nofollow">External link</a>
       #
+      # ==== Turbo
+      #
+      # Rails 7 ships with Turbo enabled by default. The <tt>data-turbo-method</tt> HTML attribute
+      # can be used to perform a Turbo link visit with a different HTTP method.
+      # {Consult the Turbo Handbook for more information on performing visits with different
+      # methods.}[https://turbo.hotwired.dev/handbook/drive#performing-visits-with-a-different-method]
+      #
+      # A common use case is to insert links to perform update or delete operations
+      # (i.e. non-<tt>GET</tt> requests) in a form.
+      #
+      # For example, you might use +button_to+ in an edit form to delete a record:
+      #
+      #   button_to "Delete profile", @profile, method: :delete
+      #
+      # Since forms cannot appear inside another form,
+      # +button_to+ will not work here as it returns a form.
```

Comment:
Is still find this whole section confusing. The bit about putting links inside forms being the most common usage doesn't sound right to me - is that actually the primary use case for this feature?

Reply:
@ghiculescu , you're right that it's likely not common to put links in forms. 

I've updated the docs to be much more concise when documenting the `data-turbo-method` and `data-turbo-confirm` options. What do you think of this change?

https://github.com/rails/rails/blob/6a798d079a875a5e60ace20ead23bb3e89eec712/actionview/lib/action_view/helpers/url_helper.rb#L176-L197